### ISSUE TITLE
Better debugging support

### ISSUE THREAD

**Issue by yissachar** [ahome-it/ahome-titanium#4]
Is there a good way to debug the generated code running on device/simulator? Currently I just get a stacktrace from the final generated JS file. This is OK, but it would be better if the code could be matched up to the original GWT source, using source maps or some other approach. I don't know how feasible this is, since by the time it's running on the device Titanium4J is out of the picture.

**Comment by eemi2010** [comment]
Debugging is really important to us. We had a debugger that was based on the old devmode protocol. But since that s getting deprecated we are looking into hooking in chrome devtools. We cant promise nothing for now but we are making progress.

Stay tuned :)

**Comment by yissachar** [comment]
Sounds great, looking forward!

**Comment by eemi2010** [comment]
Thx :) Wondering how have you experience with ti4j so far.

**Comment by yissachar** [comment]
Really impressed so far. Being able to write native apps using GWT is very powerful. We were able to get a HelloWorld up and running very quickly. Connecting that to our APIs proved not to be much more difficult. Now we are testing to see how much code we can share with a GWT web version of the app (we know that we can just generate a mobileweb version with Titanium, but we would prefer not to use that).

So far the main pain points have been debugging and refresh cycle (we haven't been able to get live reload working for some reason).

Great job :+1:

**Comment by eemi2010** [comment]
Thanks. 
Truly appreciate. We are glad that the library is helping you .
Are you guyz using Android and Windows. 
I know there are some issues with titanium shadow on windows.
I m using a mac and everything is going pretty well here.
Are you getting any errors ?

**Comment by yissachar** [comment]
We are using Mac and testing on both iOS simulator and Android emulator. I followed the instructions here http://titanium4j.ahome-it.com/livereload/ but it doesn't do anything for me. Recompiling code doesn't refresh on the device, and I have to a full rebuild from Titanium CLI.

**Comment by eemi2010** [comment]
This is really strange have u tried to see if live reload work with a regular titanium project.
Maybe Tishadow did not install properly

**Comment by yissachar** [comment]
You are correct - Tishadow didn't install properly. I didn't realize at the time, but it failed to install because I was running Node 0.12 and it required 0.10. I used nvm to grab 0.10 and reinstall Tishadow, and now I have full working live reloads with Titanium4J. This is amazing!

**Comment by eemi2010** [comment]
Super happy you got it to work. Indeen live reload is pretty impressive. And with the debugger that we working on even more impressive. We cant wait to share it soon.

We are also working on integrating with the Alloy framework. Stay tuned :)
